// @docusaurus/plugin-ideal-image/lib/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const tslib_1 = require("tslib");
const path_1 = tslib_1.__importDefault(require("path"));
function default_1(_context, options) {
    const isProd = process.env.NODE_ENV === 'production';
    return {
        name: 'docusaurus-plugin-ideal-image',
        getThemePath() {
            return path_1.default.resolve(__dirname, './theme');
        },
        configureWebpack(_config, isServer) {
            return {
                mergeStrategy: {
                    'module.rules': 'prepend',
                },
                module: {
                    rules: [
                        {
                            test: /\.(png|jpe?g|gif)$/i,
                            use: [
                                require.resolve('@docusaurus/lqip-loader'),
                                {
                                    loader: require.resolve('@docusaurus/responsive-loader'),
                                    options: {
                                        emitFile: !isServer,
                                        disable: !isProd,
                                        adapter: require('@docusaurus/responsive-loader/sharp'),
                                        name: isProd
                                            ? 'assets/ideal-img/[name].[hash:hex:7].[width].[ext]'
                                            : 'assets/ideal-img/[name].[width].[ext]',
                                        ...options,
                                    },
                                },
                            ],
                        },
                    ],
                },
            };
        },
    };
}
exports.default = default_1;
